.github/workflows/test.yml

```diff
@@ -34,30 +34,40 @@ jobs:
       - name: Test
         run: python -m pytest -v -m "not ctest"
 
-  build-openassetio:
-    name: Build OpenAssetIO
-    runs-on: ubuntu-latest
-    container:
-      image: ghcr.io/openassetio/openassetio-build
-    steps:
-      - uses: actions/checkout@v3
-      - name: Build
-        uses: ./.github/build_openassetio
-
   ctest:
-    name: C++ tests
-    runs-on: ubuntu-latest
-    needs: build-openassetio
-    container:
-      image: aswf/ci-vfxall:2022-clang14.3
+    name: C++ tests ${{ matrix.config.os }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - os: windows-2019
+            site-packages: Lib/site-packages
+          - os: ubuntu-20.04
+            site-packages: lib/python3.9/site-packages
+          - os: macos-11
+            site-packages: lib/python3.9/site-packages
+    defaults:
+      run:
+        shell: bash
     steps:
       - uses: actions/checkout@v3
 
-      - name: Get OpenAssetIO
-        uses: actions/download-artifact@v3
+      - uses: actions/setup-python@v4
         with:
-          name: OpenAssetIO Build
-          path: ${{ github.workspace }}/openassetio
+          python-version: 3.9
+
+      - name: Download Latest OpenAssetIO Artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          name: openassetio-${{ matrix.config.os }}
+          repo: openassetio/openassetio
+          workflow: build.yml
+          branch: main
```

Comment:
[fix] Can we point this to the same tag as we're installing from PyPI? I think the intention is to make sure TraitGen itself is stable during breaking changes to `main`, and dealing with those in `integrations.yml`. Otherwise, we end up with c++ and Python testing against different code bases. We might need to invoke a run against that sha? See code comment above too as its out of date if not.

Reply:
Yeah you're right, I lost sight of the intent of this, thought bleeding edge builds were the thing to do but they're not, we want a pin.

Reply:
Pinned this to a SHA here : https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/56/commits/269896260d963bec59199d9c283ea5af90cabf74

As discussed, we cant pin this to the beta yet as it didn't have builds, and it looks like we cant retroactively run actions.
(There's a couple of additional fixups dealing with CI fluff and extending the comment a bit after this)

Reply:
Seems like a good compromise to me - thanks for looking into it all

Reply:
Added  https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/issues/58